tensorflow/c/c_api_experimental.cc

```diff
@@ -743,3 +746,26 @@ void TF_ImportGraphDefOptionsSetValidateColocationConstraints(
     TF_ImportGraphDefOptions* opts, unsigned char enable) {
   opts->opts.validate_colocation_constraints = enable;
 }
+
+TF_Library* TF_LoadPluggableDeviceLibrary(const char* library_filename,
+                                          TF_Status* status) {
+#if defined(IS_MOBILE_PLATFORM) || defined(IS_SLIM_BUILD)
+  status->status = tensorflow::errors::Unimplemented(
+      "PluggableDevice plugin functionality is not supported on mobile");
+  return nullptr;
+#else
+  TF_Library* lib_handle = new TF_Library;
+  status->status = tensorflow::LoadPluggableDeviceLibrary(
```

Comment:
I think it might be cleaner to put device plugin loading logic here. Basically, have this function just call `tensorflow::LoadLibrary` and do anything else needed to load pluggable device here.

Then, `LoadPluggableDeviceLibrary` can be deleted.

Reply:
do you mean direct call `env::LoadDynamicLibrary`(not `tensorflow::LoadLibrary`) in c_api_experimental.cc? I didn't find the `tensorflow::LoadLibrary` definition, but only found  `tensorflow::LoadDynamicLibrary` in the framework/load_library.cc, however, it not only load the library, but also do the op registration.

Reply:
Yep, sorry for the typo, I meant `LoadDynamicLibrary`. I was looking at this function: tensorflow/core/platform/load_library.h which doesn't seem to load ops: https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/platform/default/load_library.cc;l=26?q=tensorflow%20LoadDynamicLibrary (unless I am looking at the wrong file).

But even if we call this implementation and register ops, I think thats ok:
https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/framework/load_library.cc;l=46?q=tensorflow%20load_library.cc
This will only run once and then cache and we call `_ll.load_library(_main_dir)` in `api_template.__init__.py` anyway. I see there would still be memcpy, don't know if that would impact latency significantly.

Reply:
done, I use env::LoadDynamicLibrary and removed LoadPluggableDeviceLibrary